src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
+}
```

Comment:
Also, I hadn't thought about it until you mentioned it, but in express you can call `next` without returning it. I'm wondering if (in spite of the clarity of the readme) this might be a source of confusion for people coming from express?